src/detect.c

```diff
@@ -813,24 +813,22 @@ static inline void DetectRulePacketRules(
         DetectRunPostMatch(tv, det_ctx, p, s);
 
         uint64_t txid = PACKET_ALERT_NOTX;
-        if ((alert_flags & PACKET_ALERT_FLAG_STREAM_MATCH) ||
-                (s->alproto != ALPROTO_UNKNOWN && pflow->proto == IPPROTO_UDP)) {
+        if (pflow && pflow->alstate &&
+                ((alert_flags & PACKET_ALERT_FLAG_STREAM_MATCH) ||
+                        (s->alproto != ALPROTO_UNKNOWN && pflow->proto == IPPROTO_UDP) ||
+                        AppLayerParserGetTxCnt(pflow, pflow->alstate) == 1)) {
             // if there is a stream match (TCP), or
             // a UDP specific app-layer signature,
+            // or only one transaction
```

Comment:
I think this is a bit vague. The point is that there is only one tx over the lifetime of the flow (so far), so not just one "active" tx. Think the git message can be more clear on this as well.

Reply:
I think this code checks if there is one active transaction, not just one over the whole lifetime...

Am I wrong ?

Reply:
Fixed this in next version